gdraw/gbuttons.c

```diff
@@ -992,10 +993,14 @@ return;
     _GGadget_droplist_box.border_shape = _ggadget_Default_Box.border_shape;
 #endif
     _GGadget_colorbutton_box  = _GGadget_button_box;
-    _GGadgetInitDefaultBox("GDefaultButton.",&_GGadget_defaultbutton_box,NULL);
-    _GGadgetInitDefaultBox("GCancelButton.",&_GGadget_cancelbutton_box,NULL);
-    _GGadgetInitDefaultBox("GDropList.",&_GGadget_droplist_box,NULL);
-    _GGadgetInitDefaultBox("GColorButton.",&_GGadget_colorbutton_box,NULL);
+    tmp = _GGadgetInitDefaultBox("GDefaultButton.",&_GGadget_defaultbutton_box,NULL);
+    free(tmp);
+    tmp = _GGadgetInitDefaultBox("GCancelButton.",&_GGadget_cancelbutton_box,NULL);
+    free(tmp);
+    tmp = _GGadgetInitDefaultBox("GDropList.",&_GGadget_droplist_box,NULL);
+    free(tmp);
+    tmp = _GGadgetInitDefaultBox("GColorButton.",&_GGadget_colorbutton_box,NULL);
+    free(tmp);
```

Comment:
How does this make sense? What's the point of initializing it at all if you're just going to free it immediately after?

Reply:
I thought that it was to initialize _GGadget_colorbutton_box but _GGadgetInitDefaultBox is more complicated than I thought.
What annoys me is this line:
 fi = GDrawInstanciateFont(NULL,&rq); at the end of the function.
if the caller doesn't retrieve the font, it can leak here.

Reply:
@jtanx : your bt may indeed indicates that my changes with functions calling _GGadgetInitDefaultBox are the culprits, what do you think about it and our last comments with ctrlcctrlv?